backends/dpdk/dpdkArch.cpp

```diff
@@ -2448,4 +2446,86 @@ const IR::Node* ElimHeaderCopy::postorder(IR::Member* m) {
     return m;
 }
 
+const IR::Node* DoFlattenHeaderUnion::postorder(IR::Type_Struct* s) {
+    auto fields = new IR::IndexedVector<IR::StructField>;
+    for (auto sf : s->fields) {
+        auto ftype = typeMap->getType(sf, true);
+        if (ftype->is<IR::Type_HeaderUnion>()) {
+            for (auto sfu : ftype->to<IR::Type_HeaderUnion>()->fields) {
+                cstring uName = sf->name.name + "_" + sfu->name.name;
+                auto uType = sfu->type->getP4Type();
+                fields->push_back(new IR::StructField(IR::ID(uName), uType));
+            }
+        } else {
+            fields->push_back(sf);
+        }
+    }
+    return new IR::Type_Struct(s->name, s->annotations, *fields);
+}
+
+const IR::Node* DoFlattenHeaderUnion::postorder(IR::Declaration_Variable *dv) {
+    auto ftype = typeMap->getTypeType(dv->type, true);
+    if (ftype->is<IR::Type_HeaderUnion>()) {
+        for (auto sfu : ftype->to<IR::Type_HeaderUnion>()->fields) {
+             cstring uName = dv->name.name + "_" + sfu->name.name;
+             auto uType =  sfu->type->getP4Type();
+             toInsert.push_back(new IR::Declaration_Variable(IR::ID(uName), uType));
+        }
+    }
+    return dv;
+}
+
+const IR::Node* DoFlattenHeaderUnion::postorder(IR::Member* m) {
+    if (m->expr->type->to<IR::Type_HeaderUnion>()) {
+        if (auto huf = m->expr->to<IR::Member>()) {
+            return new IR::Member(huf->expr,IR::ID(huf->member.name+"_"+m->member.name));
+        } else if (auto huf = m->expr->to<IR::PathExpression>()) {
+            return new IR::PathExpression(IR::ID(huf->path->name.name + "_" + m->member.name));
```

Comment:
don't you need member.originalName here?

Reply:
No. I need the unique internal names. Original name might result in name conflict for variables coming from different scope